### PR TITLE
MINOR: [R] Add note about the deprecated `type()` function

### PR DESCRIPTION
**Pull request by eitsupi**
The `type()` function was deprecated in #12817 (ARROW-15168), but the documentation did not indicate that it had been deprecated.

### CODE REVIEW

**Review by assignUser — APPROVED** [review]
LGTM, thanks!

File: r/R/type.R
Comment (nealrichardson) [inline r/R/type.R:61]: Is this line intentionally duplicated?

Comment (thisisnic) [reply r/R/type.R:61]: Somehow missed this before I merged the PR; I think it's pretty safe to say that this is a typo, so I'll open a follow-up PR.

Comment (eitsupi) [reply r/R/type.R:61]: This is intentional.
The first line is used as the title (so it does not end in a comma) and the second line is the body.

https://github.com/eitsupi/arrow/blob/67a7596f43098d5f899547b47332f6e710a5036c/r/man/infer_type.Rd#L6-L22

Comment (thisisnic) [reply r/R/type.R:61]: https://github.com/apache/arrow/pull/13822

Comment (thisisnic) [reply r/R/type.R:61]: Ah, my mistake, sorry for assuming it was a typo there!  I'm going to remove the duplication in another PR - I see where you're coming from about having a title and description, but it's unnecessary to duplicate it and not consistent with other function docs in the package.

Comment (thisisnic) [reply r/R/type.R:61]: That's just how pkgdown renders it if there is a title, but no description - see the Roxygen header for `read_schema()` versus how it appears on the website for another example of this.

Comment (eitsupi) [reply r/R/type.R:61]: @thisisnic This has nothing to do with pkgdown, it is the behavior of roxygen2.
Please check the Rd file diff of this Pull Request.
My intention was to keep the original Rd file as intact as possible. It is sad that this intention was not communicated and was considered an inappropriate update.

BTW you seem to have forgotten to update the Rd file in #13824.
This needs to be corrected.

Comment (thisisnic) [reply r/R/type.R:61]: @eitsupi No worries about the crossed wires - FWIW it's not a priority to keep the Rd file looking the same.  Thanks again for this contribution :)

Comment (eitsupi) [reply r/R/type.R:61]: @thisisnic https://arrow.apache.org/docs/developers/reviewing.html says:

> Arrow is a foundational project that will need to evolve over many years or even decades, while serving potentially millions of users. We believe that being meticulous when reviewing brings greater rewards to the project than being lenient and aiming for quick merges.

> Reviewing is a communication between the contributor and the reviewer. Avoid letting questions or comments remain unanswered for too long (“too long” is of course very subjective, but two weeks can be a reasonable heuristic). If you cannot allocate time soon, do say it explicitly. If you don’t have the answer to a question, do say it explicitly. Saying “I don’t have time immediately but I will come back later, feel free to ping if I seem to have forgotten” or “Sorry, I am out of my depth here” is always better than saying nothing and leaving the other person wondering.

You made the change without waiting for my reply (or ignoring my reply), did you not?

I hope you will act as a matter of principle in the future. (Or I may have misunderstood something. In that case, I am very sorry.)

Comment (thisisnic) [reply r/R/type.R:61]: @eitsupi This PR is for a minor change, which I was keen to get merged quickly as it's pretty straightforward.  I already apologised above for assuming that the incorrect addition was a typo.  I am very familiar with those guidelines, having helped review and update the phrasing of them when they were being drafted.  

I can only assume from your comment about me ignoring your reply (I did not  - things happen out of sync when you have multiple browser windows open), and the way that this conversation has continued over what appears to be a pretty trivial misunderstanding, that you believe I have a problem with you or your PRs.  This is not the case - honestly, it's great to have more people involved in the project, and it's really helpful to have more eyes on things like docs which are important but easily missed.  

That said, it's clear that our interactions are not a positive experience for either of us, and so in future I'm going to avoid reviewing your PRs so that they can instead be reviewed by other members of the community.

Comment (eitsupi) [reply r/R/type.R:61]: @thisisnic My point is that you forgot to update the Rd file in #13822 or #13824 and merged #13824 without waiting for my reply, so this "pretty straightforward" problem is still not fixed even though a week has passed.

Comment (thisisnic) [reply r/R/type.R:61]: @eitsupi I'm done here.

Comment (thisisnic) [reply r/R/type.R:61]: @eitsupi I'm not sure exactly why you are continuing to argue with me about a trivial issue which has no practical consequences, but I'm done here.

**Pull request closed without merging**
